Client/Frontend/Settings/LoginDetailViewController.swift

```diff
@@ -379,9 +379,15 @@ extension LoginDetailViewController {
             return
         }
 
-        let updated = Login(guid: login.guid, hostname: login.hostname, username: username, password: password)
-        if updated.isValid.isSuccess {
+        // Keep a copy of the old data in case we fail and need to revert back
+        let oldPassword = login.password
+        let oldUsername = login.username
+        login.updateWithPassword(password, username: username)
+
+        if login.isValid.isSuccess {
             profile.logins.updateLoginByGUID(login.guid, new: login, significant: true)
+        } else if let oldUsername = oldUsername {
+            login.updateWithPassword(oldPassword, username: oldUsername)
```

Comment:
After the failure, we want to reset back to the original state the login was (for now). This isn't the best UX for telling the user that the login was invalid but works for 2.0.